### DISCUSSION TITLE
What is the possibility of dynamically showing the NAPLAM related tabs?

### DISCUSSION THREAD

**Discussion by WhaleJ84** [netbox-community/netbox#6025] [Ideas]
Hey guys,

On every device you have the `Status`, `LLDP Neighbors`, and `Configuration` tabs, which after hovering over the tabs the helpful tool-tip sends you on an adventure.

 - `No platform assigned to this device` > go assign it a platform
 - `Device must be in active status` > set it to active
 - Probably some other steps I haven't found
 
 And then finally:
 - `No NAPALM driver assigned for this platform` > go find out what platforms have NAPALM drivers
 
Here is where people find that NAPALM (at least at the time of posting) is only for networking devices, and even then only for [a very small number of manufacturers!](https://napalm.readthedocs.io/en/latest/support/)
What this means is that a very large majority of people will **never** use the NAPALM drivers and be doomed to have 3 unusable tabs ahead of the `Config Context` and `Change Log` tabs.

We know that those tabs can be dynamically assigned per device as `Interfaces`, `Power Ports`, and `Power Outlets` (to name a few) will only show on devices where they have been assigned.
Is there no way to do the same for devices that are Active, have a platform assigned AND have the NAPALM driver for it available?

I've looked around to see if this topic has been touched on before but it did not appear, so apologies in advance if I have missed it.

I'm mostly looking to see what other peoples thoughts are on this and if it should be converted into an issue for future consideration but this my first time using the discussions so I'm interested to see how it will go!

**Comment by rodvand** [comment]
I have been thinking about the same thing. Using 3/7 tabs for this feature seems wasteful. Maybe it makes more sense to move the NAPALM functionality to a plugin (and maybe re-visit https://github.com/netbox-community/netbox/issues/5556 to allow plugins to add tabs on device/VM pages).

**Comment by jeremystretch** [comment]
IIRC, these tabs were originally always displayed to both call attention to the feature and to assist in troubleshooting (hence the error messages). As you argue, there's probably no need to always display them anymore. IMO it makes sense to display these tabs only when all of the cited conditions have been met. However, I do think we should keep them as discrete tabs to stay consistent with the rest of the UI.

**Reply by WhaleJ84** [reply]
I also agree there.
I get that this opinion has only been echoed by a very small minority of the user-base and it's in no way something that would benefit the progression of Netbox, but is there anything more that would be needed for this to be added into the roadmap?

**Reply by jeremystretch** [reply]
You're welcome to open an FR for it so that it gets more attention. I don't expect any pushback.

**Comment by WhaleJ84** [comment]
Opened [issue #6054](https://github.com/netbox-community/netbox/issues/6054)
